### PR TITLE
Add proper description for `Rails.cache.fetch` [ci skip]

### PR DESCRIPTION
**Pull request by mechanicles**
Add proper description for `Rails.cache.fetch`.

### CODE REVIEW

File: guides/source/caching_with_rails.md [hunk 1]
```diff
@@ -270,7 +270,7 @@ simply be explicit in a comment, like:
 
 Sometimes you need to cache a particular value or query result instead of caching view fragments. Rails' caching mechanism works great for storing __any__ kind of information.
 
-The most efficient way to implement low-level caching is using the `Rails.cache.fetch` method. This method does both reading and writing to the cache. When passed only a single argument, the key is fetched and value from the cache is returned. If a block is passed, the result of the block will be cached to the given key and the result is returned.
+The most efficient way to implement low-level caching is using the `Rails.cache.fetch` method. This method does both reading and writing to the cache. When passed only a single argument, the key is fetched and value from the cache is returned. If a block is passed for given key which is not used for cache yet, then result of the block will be cached to that key and the result is returned otherwise value from the cache for that key is returned.
```
Comment (prathamesh-sonpatki) [inline guides/source/caching_with_rails.md]: What do you think about ~

```
If a block is passed, that block will be executed in the event of a cache miss. The return
value of the block will be written to the cache under the given cache key, and that return value will be returned.
```

Comment (mechanicles) [inline guides/source/caching_with_rails.md]: @prathamesh-sonpatki Some new developers can not digest `cache miss` & `cache hit` terms easily on the first impression. Need to mention `cache key` explicitly I think. All other parts look good to me 👍 

How about this one (changed little bit)

```
If a block is passed, that block will be executed when given cache key is not found. The return value of the block will be written to the cache under the given cache key, and that return value will be returned. If cache key is found, then the cached value will be returned without executing the block.
```

Comment (kaspth) [inline guides/source/caching_with_rails.md]: > Some new developers can not digest cache miss & cache hit terms easily on the first impression

Cache miss & hit is common terminology when talking about caching and, I think, fair game in documentation. I'd rather we just use those than end up using more words to explain a common concept. 😁

Comment (mechanicles) [inline guides/source/caching_with_rails.md]: Got it 👍

**Comment by mechanicles** [comment]
@prathamesh-sonpatki @kaspth Updated the changes based on your suggestions. Please check.

**Comment by mechanicles** [comment]
@prathamesh-sonpatki btw Thanks for good documentation help :)

**Comment by prathamesh-sonpatki** [comment]
Thanks!

**Pull request merged**